client/pchainclient.go

```diff
@@ -0,0 +1,93 @@
+	"github.com/ava-labs/avalanchego/vms/avm"
+	"github.com/ava-labs/avalanchego/vms/platformvm"
+	"github.com/ava-labs/avalanchego/vms/platformvm/signer"
+)
+
+// Interface compliance
+var _ PChainClient = &pchainClient{}
+
+type PChainClient interface {
+	// indexer.Client methods
+	GetContainerByIndex(ctx context.Context, index uint64, options ...rpc.Option) (indexer.Container, error)
+	GetContainerByID(ctx context.Context, containerID ids.ID, options ...rpc.Option) (indexer.Container, error)
+	GetLastAccepted(context.Context, ...rpc.Option) (indexer.Container, error)
+
+	// platformvm.Client methods
+
+	GetUTXOs(
+		ctx context.Context,
+		addrs []ids.ShortID,
+		limit uint32,
+		startAddress ids.ShortID,
+		startUTXOID ids.ID,
+		options ...rpc.Option,
+	) ([][]byte, ids.ShortID, ids.ID, error)
+	GetAtomicUTXOs(
+		ctx context.Context,
+		addrs []ids.ShortID,
+		sourceChain string,
+		limit uint32,
+		startAddress ids.ShortID,
+		startUTXOID ids.ID,
+		options ...rpc.Option,
+	) ([][]byte, ids.ShortID, ids.ID, error)
+	GetRewardUTXOs(context.Context, *api.GetTxArgs, ...rpc.Option) ([][]byte, error)
+	GetHeight(ctx context.Context, options ...rpc.Option) (uint64, error)
+	GetBalance(ctx context.Context, addrs []ids.ShortID, options ...rpc.Option) (*platformvm.GetBalanceResponse, error)
+	GetTx(ctx context.Context, txID ids.ID, options ...rpc.Option) ([]byte, error)
+	GetBlock(ctx context.Context, blockID ids.ID, options ...rpc.Option) ([]byte, error)
+	IssueTx(ctx context.Context, tx []byte, options ...rpc.Option) (ids.ID, error)
+	GetStake(ctx context.Context, addrs []ids.ShortID, options ...rpc.Option) (map[ids.ID]uint64, [][]byte, error)
+
+	// avm.Client methods
+
+	GetAssetDescription(ctx context.Context, assetID string, options ...rpc.Option) (*avm.GetAssetDescriptionReply, error)
+
+	// info.Client methods
+	IsBootstrapped(context.Context, string, ...rpc.Option) (bool, error)
+	Peers(context.Context, ...rpc.Option) ([]info.Peer, error)
+	GetNodeID(context.Context, ...rpc.Option) (ids.NodeID, *signer.ProofOfPossession, error)
+	GetNetworkID(context.Context, ...rpc.Option) (uint32, error)
+	GetBlockchainID(context.Context, string, ...rpc.Option) (ids.ID, error)
+	GetTxFee(context.Context, ...rpc.Option) (*info.GetTxFeeResponse, error)
+}
+
+type (
+	indexerClient    = indexer.Client
+	platformvmClient = platformvm.Client
+	infoClient       = info.Client
+)
+
+type pchainClient struct {
+	platformvmClient
+	indexerClient
+	infoClient
+	xChainClient avm.Client
+}
+
+// NewPChainClient returns a new client for Avalanche APIs related to P-chain
+func NewPChainClient(ctx context.Context, endpoint, indexerEndpoint string) PChainClient {
+	endpoint = strings.TrimSuffix(endpoint, "/")
+
+	return pchainClient{
+		platformvmClient: platformvm.NewClient(endpoint),
+		xChainClient:     avm.NewClient(endpoint, "X"),
+		infoClient:       info.NewClient(endpoint),
+		indexerClient:    indexer.NewClient(indexerEndpoint + "/ext/index/P/block"),
```

Comment:
Related to the later comment on how this flag is passed, but this looks like `indexerEndpoint` is basically expected to just be the raw endpoint. I think it would make sense to either use the `endpoint` instead or to have `indexerEndpoint` be expected to already have this suffix.

Reply:
Having `indexerEndpoint` separate from `endpoint` allows us to point to public indexer endpoint while pointing endpoint to the public api endpoint which are separate.

Since the suffix can potentially be changed in future versions as part of a node update, I think having it here instead of let it come from the configuration would allow silently changing it in code instead of needing users to update their configs.